src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -488,3 +497,27 @@ def _get_list_runs_message(base_message, task_name=None, image=None):
     if image:
         base_message = "{} for image '{}'".format(base_message, image)
     return "{}.".format(base_message)
+
+
+def _get_trigger_event_list(source_triggers,
+                            commit_trigger_enabled=None,
+                            pull_request_trigger_enabled=None):
+    source_trigger_events = set()
+    # perform merge with server-side event list
+    if source_triggers:
+        source_trigger_events = set(source_triggers[0].source_trigger_events)
+        if source_triggers[0].status == TriggerStatus.disabled.value:
+            source_trigger_events.clear()
```

Comment:
This also relates to the above question. Is it possible that `source_trigger_events` is not empty (so you are clearing it) but `status` is disabled?

Reply:
There is an edge case in the RP that checks if ```source_trigger_events``` is empty. During update, if the user disabled all events in a trigger and sends an empty ```source_trigger_events``` to the RP, then the RP overrides these events with those of the existing trigger. 

```
            sourceTrigger.SourceTriggerEvents = sourceTrigger.SourceTriggerEvents?.Count() > 0 ?
                sourceTrigger.SourceTriggerEvents : 
                existingTrigger.SourceTriggerEvents;
```

Therefore, a disabled trigger event list may be "empty" to the client it is not to the server (though the trigger is still disabled). This ```.clear()``` is here because if a user is calling update again on a trigger that has been disabled, then we want to make sure none of the previously disabled events events are added back without the user knowing.